src/main/java/com/readup/server/auth/infrastructure/KeyManager.java

```diff
@@ -0,0 +1,31 @@
+package com.readup.server.auth.infrastructure;
+
+import java.nio.charset.StandardCharsets;
+import java.security.Key;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.stereotype.Component;
+
+import io.jsonwebtoken.security.Keys;
+import jakarta.annotation.PostConstruct;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+@Component
+@ConfigurationProperties(prefix = "spring.security.oauth2.keys")
+public class KeyManager {
+
+	private String accessTokenKeyString;
+	private String refreshTokenKeyString;
+
+	private Key accessTokenKey;
+	private Key refreshTokenKey;
+
+	@PostConstruct
+	public void init() {
+		this.accessTokenKey = Keys.hmacShaKeyFor(accessTokenKeyString.getBytes(StandardCharsets.UTF_8));
+		this.refreshTokenKey = Keys.hmacShaKeyFor(refreshTokenKeyString.getBytes(StandardCharsets.UTF_8));
+	}
+}
```

Comment:
```suggestion
@Getter
@ConfigurationProperties(prefix = "spring.security.oauth2.keys")
public class KeyManager {

	private Key accessTokenKey;
	private Key refreshTokenKey;

	@ConstructorBinding
	public KeyManager(String accessTokenKeyString, String refreshTokenKeyString) {
		this.accessTokenKey = Keys.hmacShaKeyFor(accessTokenKeyString.getBytes(StandardCharsets.UTF_8));
		this.refreshTokenKey = Keys.hmacShaKeyFor(refreshTokenKeyString.getBytes(StandardCharsets.UTF_8));
	}
}
```

이런 방식은 어떠실까요? accessTokenKeyString과 refreshTokenKeyString이 사용되지 않는다면 좋은 방법이지 않을까 싶어요! 작동이 되는지는 확인해봐야 합니다만, 저도 테스트 해보겠습니다!

추가적으로 ServerApplication 에 @ConfigurationPropertiesScan 이 있어서 @Component 가 없어도 Bean으로 자동 등록 되는 것으로 알고 있습니다!

Reply:
@ConfigurationPropertiesScan  이 부분은 제가 간과하고 있었네요 @Component는 삭제 했습니다.
넵 위와 같이 작성하는게 깔끔해보입니다! 수정했습니다.